### PR TITLE
Implemented Primitive

### PR DESCRIPTION
**Pull request by daeyounglnc**
#### What this PR does / why we need it?
Implemented `Primitive` and it's testcode. 
The description of this class : [js-sdk doc](https://yorkie.dev/yorkie-js-sdk/yorkie-js-sdk.html)

#### Any background context you want to provide?
This PR doesn't contain fully spec of Primitive. 
It will be implemented later when it needed.

#### What are the relevant tickets?
None

Fixes #

### Checklist

- [x] Added relevant tests or not required
- [x] Didn't break anything

### CODE REVIEW

File: yorkie/src/main/kotlin/dev/yorkie/document/crdt/Primitive.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    init {
+        type = when (value) {
+            is Boolean -> PrimitiveType.Boolean
+            is Int -> PrimitiveType.Integer
+            is Long -> PrimitiveType.Long
+            is Double -> PrimitiveType.Double
+            is String -> PrimitiveType.String
+            is ByteArray -> PrimitiveType.Bytes
+            // TODO support Date type
```
Comment (7hong13) [inline yorkie/src/main/kotlin/dev/yorkie/document/crdt/Primitive.kt:19]: @hackerwins should we write `TODO` in the form below?
```suggestion
TODO(daeyoungInc): support Date type
```
(ref: https://github.com/yorkie-team/yorkie-android-sdk/pull/7#discussion_r972191488)

File: yorkie/src/main/kotlin/dev/yorkie/document/crdt/Primitive.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    init {
+        type = when (value) {
+            is Boolean -> PrimitiveType.Boolean
+            is Int -> PrimitiveType.Integer
+            is Long -> PrimitiveType.Long
+            is Double -> PrimitiveType.Double
+            is String -> PrimitiveType.String
+            is ByteArray -> PrimitiveType.Bytes
+TODO(daeyoungInc): support Date type
+            else -> PrimitiveType.Null
```
Comment (hackerwins) [inline yorkie/src/main/kotlin/dev/yorkie/document/crdt/Primitive.kt:20]:
```suggestion
            is ByteArray -> PrimitiveType.Bytes
            // TODO(daeyoungInc): support Date type
            else -> PrimitiveType.Null
```

File: yorkie/src/main/kotlin/dev/yorkie/document/crdt/Primitive.kt [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    init {
+        type = when (value) {
+            is Boolean -> PrimitiveType.Boolean
+            is Int -> PrimitiveType.Integer
+            is Long -> PrimitiveType.Long
+            is Double -> PrimitiveType.Double
+            is String -> PrimitiveType.String
+            is ByteArray -> PrimitiveType.Bytes
+            // TODO(daeyoungInc): support Date type
```
Comment (hackerwins) [inline yorkie/src/main/kotlin/dev/yorkie/document/crdt/Primitive.kt:19]:
```suggestion
            // TODO(daeyounglnc): support Date type
```

**Comment by hackerwins** [comment]
@daeyounglnc FYI, Although Firebase Android SDK uses Java, it seems to use Date type for Document.

https://github.com/firebase/firebase-android-sdk/blob/ac3276ce7d5f5181b1c419731495eb88a98537af/firebase-firestore/src/androidTest/java/com/google/firebase/firestore/TypeTest.java#L110

**Pull request merged**